packages/rum-core/src/common/apm-server.js

```diff
@@ -80,39 +80,51 @@ class ApmServer {
     })
   }
 
-  _makeHttpRequest(method, url, payload, headers) {
+  _makeHttpRequest(method, url, payload, headers = {}) {
     return new Promise(function(resolve, reject) {
       var xhr = new window.XMLHttpRequest()
       xhr[XHR_IGNORE] = true
       xhr.open(method, url, true)
       xhr.timeout = 10000
 
-      if (headers) {
-        for (var header in headers) {
-          if (headers.hasOwnProperty(header)) {
-            xhr.setRequestHeader(header, headers[header])
-          }
+      for (var header in headers) {
+        if (headers.hasOwnProperty(header)) {
+          xhr.setRequestHeader(header, headers[header])
         }
       }
 
       xhr.onreadystatechange = function() {
         if (xhr.readyState === 4) {
-          var status = xhr.status
+          const { status, responseText } = xhr
+          // An http 4xx or 5xx error. Signal an error.
           if (status === 0 || (status > 399 && status < 600)) {
-            // An http 4xx or 5xx error. Signal an error.
-            var err = new Error(url + ' HTTP status: ' + status)
-            err.xhr = xhr
+            let message = []
```

Comment:
I think this should be moved to the calling method where e.g. [here](https://github.com/elastic/apm-agent-rum-js/blob/master/packages/rum-core/src/common/apm-server.js#L135). 

is a generic method and should only resolve or reject the promise.

Reply:
Not really, the only job of the method is to construct the error based on status code and we cannot replicate the error creation logic multiple times since we have error handling logic for transaction/error.

Reply:
We don't have to duplicate the error creation we can put it in a function and call that, but we should call that function in the caller of `_makeHttpRequest`